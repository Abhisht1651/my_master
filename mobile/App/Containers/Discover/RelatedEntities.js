// @flow

import React from 'react'
import { View, ScrollView, Image, Text, TouchableOpacity, Platform, ImageProps } from 'react-native'
import { NavigationContainerProps } from '@react-navigation/native'

// Styles
import styles from '../Styles/Discover/RelatedEntitiesStyle'
import images from '../../Themes/Images'

// Components
import Entities from '../../Components/Entities/CardType1'

type Props = NavigationContainerProps & {
  headerTitle: string,
  Data: Array<{
    title: string,
    count: number,
    profileIcons: ImageProps
  }>
}

const relatedEntities = (props: Props) => {
  const { navigation, headerTitle, Data } = props
  const { backIcon } = images

  const EntitiesDataElements = Data.map((item, index) => {
    const { title, count, profileIcons } = item
    return (
      <View key={index} style={styles.previewContainer}>
        <Entities title={title} count={count} profileIcons={profileIcons} />
      </View>
    )
  })
  return (
    <View style={styles.container}>
      {Platform.OS === 'ios' ? <View style={styles.topCover} /> : null}
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Image source={backIcon} style={styles.backIcon} />
        </TouchableOpacity>
        <Text style={styles.relatedText}>{headerTitle}</Text>
      </View>
      <ScrollView showsVerticalScrollIndicator={false}>
        <View style={styles.scrollContainer}>{EntitiesDataElements}</View>
      </ScrollView>
    </View>
  )
}

relatedEntities.defaultProps = {
  headerTitle: 'Related Entities',
  Data: [
    {
      title: 'India',
      count: 52,
      profileIcons: [images.EntitiesImage0]
    },
    {
      title: 'United States of America',
      count: 52,
      profileIcons: [images.EntitiesImage1]
    },
    {
      title: 'India',
      count: 52,
      profileIcons: [images.EntitiesImage0]
    },
    {
      title: 'United States of America',
      count: 52,
      profileIcons: [images.EntitiesImage1]
    },
    {
      title: 'India',
      count: 52,
      profileIcons: [images.EntitiesImage0]
    },
    {
      title: 'United States of America',
      count: 52,
      profileIcons: [images.EntitiesImage1]
    },
    {
      title: 'India',
      count: 52,
      profileIcons: [images.EntitiesImage0]
    },
    {
      title: 'United States of America',
      count: 52,
      profileIcons: [images.EntitiesImage1]
    }
  ]
}

export default relatedEntities
