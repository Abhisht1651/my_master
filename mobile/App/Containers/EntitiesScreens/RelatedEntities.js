// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React from 'react'
import { View, Text, TouchableOpacity, Image, ScrollView } from 'react-native'
import images from '../../Themes/Images'
// Styles
import styles from './ScreenStyles/RelatedEntitiesStyles'
import CardType1 from '../../Components/Entities/CardType1'

type Props = {} & NavigationContainerProps

const RelatedEntities = ({ navigation }: Props) => {
  return (
    <View style={styles.screen}>
      <View style={styles.topBox}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Image source={images.arrow} />
        </TouchableOpacity>
        <Text style={styles.textFont}>Related Entities</Text>
      </View>
      <View style={styles.contentBox}>
        <ScrollView>
          <View style={styles.box1}>
            <CardType1 title="India" profileIcons={[images.EntitiesImage0]} count={52} />
          </View>
          <View style={styles.box0}>
            <CardType1
              title="United States of America"
              profileIcons={[images.EntitiesImage1]}
              count={51}
            />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
          <View style={styles.box0}>
            <CardType1 title="United States of America" />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
          <View style={styles.box0}>
            <CardType1 />
          </View>
        </ScrollView>
      </View>
    </View>
  )
}

export default RelatedEntities
