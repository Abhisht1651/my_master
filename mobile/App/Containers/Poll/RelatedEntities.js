// @flow

import React from 'react'
import { View, ScrollView, Image, Text, TouchableOpacity, Platform } from 'react-native'
import { NavigationContainerProps } from '@react-navigation/native'

// Styles
import styles from '../Styles/Poll/RelatedEntitiesStyle'
import images from '../../Themes/Images'

// Components
import Entities from '../../Components/Entities/CardType1'

type Props = NavigationContainerProps & {
  storyLineData: Array<{
    reach: number,
    heading: string,
    storyTitle: string,
    friendsFollowing: number,
    updatedTime: number,
    following: boolean,
    trending: boolean
  }>
}

const relatedStoryLine = (props: Props) => {
  const { navigation, storyLineData } = props
  const { backIcon } = images

  const storyLineDataElements = storyLineData.map((item, index) => {
    const { title, count, profileIcons } = item
    return (
      <View key={index} style={styles.previewContainer}>
        <Entities title={title} count={count} profileIcons={profileIcons} />
      </View>
    )
  })
  return (
    <View style={styles.container}>
      {Platform.OS === 'ios' ? <View style={styles.topCover} /> : null}
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Image source={backIcon} style={styles.backIcon} />
        </TouchableOpacity>
        <Text style={styles.relatedText}>Related Entities</Text>
      </View>
      <ScrollView showsVerticalScrollIndicator={false}>
        <View style={styles.scrollContainer}>{storyLineDataElements}</View>
      </ScrollView>
    </View>
  )
}

relatedStoryLine.defaultProps = {
  storyLineData: [
    {
      title: 'India',
      count: 52,
      profileIcons: [images.EntitiesImage0]
    },
    {
      title: 'United States of America',
      count: 52,
      profileIcons: [images.EntitiesImage1]
    },
    {
      title: 'India',
      count: 52,
      profileIcons: [images.EntitiesImage0]
    },
    {
      title: 'United States of America',
      count: 52,
      profileIcons: [images.EntitiesImage1]
    },
    {
      title: 'India',
      count: 52,
      profileIcons: [images.EntitiesImage0]
    },
    {
      title: 'United States of America',
      count: 52,
      profileIcons: [images.EntitiesImage1]
    },
    {
      title: 'India',
      count: 52,
      profileIcons: [images.EntitiesImage0]
    },
    {
      title: 'United States of America',
      count: 52,
      profileIcons: [images.EntitiesImage1]
    }
  ]
}

export default relatedStoryLine
