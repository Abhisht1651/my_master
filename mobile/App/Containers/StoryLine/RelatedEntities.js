// @flow

// import React from 'react'
import React, { useState, useEffect, useRef } from 'react'
import {withApollo} from 'react-apollo';
import {getEntityDetails,getRelatedEntities,getMultimediaDetails,AddEntityFollower,deleteEntityFollower,updateEntityFollowers,GetEntityFollower} from './my_queries';

import { View, ScrollView, Image, Text, TouchableOpacity, Platform, ImageProps } from 'react-native'
import { NavigationContainerProps } from '@react-navigation/native'

// Styles

import styles from '../Styles/StoryLine/RelatedEntitiesStyle'
import images from '../../Themes/Images'

// Components
import Entities from '../../Components/Entities/CardType1'

// type Props = NavigationContainerProps & {
//   Data: Array<{
//     title: string,
//     count: number,
//     profileIcons: ImageProps
//   }>
// }

const relatedEntities = (props) => {
  const user_id="5081a186-6c47-48fa-aaff-dbd5f45ee242";
  var [relatedEntitiesData1,setFeed]=useState([]);
  // var [url,seturl]=useState([]);
  const client=props.client;
  if (!client){
    console.log("CLient not working")
  }
  const onPressFollowUtil = async ({ followstate, entity_id, followers }) => {
    if (followstate) {
      followers +=1
      const { data: data1 } = await client.mutate({
        mutation: AddEntityFollower,
        variables: { entity_id, user_id }
      })
      const { data: data2 } = await client.mutate({
        mutation: updateEntityFollowers,
        variables: { entity_id: `'${entity_id}'`, followers }
      })
    } else {
      followers -= 1
      const { data: data1 } = await client.mutate({
        mutation: deleteEntityFollower,
        variables: { entity_id, user_id }
      })
      const { data: data2 } = await client.mutate({
        mutation: updateEntityFollowers,
        variables: { entity_id: `'${entity_id}'`, followers }
      })
    }
  }
  
  // const { navigation, storyLineData } = props
  const { backIcon } = images
  
  const displayrelatedentity = async (client) =>{
    try{
      let source=[];
      const entity_id_arr = [
        'ff3b2a22-8858-428e-8270-98f31ae4acdc',
        '96a4ae77-753c-4cce-9a24-37050a0cf77f',
        'e75d2349-7ff4-4a8f-aa54-1fa16a5a5082',
        'ec0dfbbd-466d-411e-8923-8fcd5c6824cd',
        '4caed58a-0fe6-4d75-90b4-dffe6bc395a7'
      ]
      for(var i=0;i<entity_id_arr.length;i++)
      {
          let item={}
          const entity_id=entity_id_arr[i];
          // const {data: datarelatedentity, loading: loadingrelatedentity} = await client.query({query: getRelatedEntities, variables: {entity_id:entity_id1}});
          // const {entity_id}=datarelatedentity.GetRelatedStorylinesToEntity[0];
          // console.log(entity_id)
          const {data: dataentity, loading: loadingentity} = await client.query({query: getEntityDetails, variables: {entity_id:entity_id}});
          const {followers,image,name}=dataentity.GetEntity;
          const multi_id=image
          const {data:datamulti,loading:loadingmulti}= await client.query({query:getMultimediaDetails,variables:{multimedia_id:multi_id}});
          const {url}=datamulti.GetMultimedia;
          const { data: dataEntityFollower, loading:lF, error:eF } = await client.query({
            query: GetEntityFollower,
            variables: { entity_id,user_id }
          })
          const is_following=dataEntityFollower.GetEntityFollower.length>0
          item["title"]=name;
          item['count']=followers;
          item['profileIcons']=[url];
          item['onPressFollow']= async (followstate) => { await onPressFollowUtil({ followstate, entity_id, followers }) }
          item['is_following']=is_following
          source[i]=item
      }
      setFeed(source);
      }
    catch(e){
      console.log(e)
      }
    }
  useEffect(() => {displayrelatedentity(client);}, []);

  const { navigation, Data } = props
  // const { backIcon } = images

  const EntitiesDataElements = relatedEntitiesData1.map((item, index) => {
    const { title, count, profileIcons,onPressFollow,is_following } = item
    return (
      <View key={index} style={styles.previewContainer}>
        <Entities title={title} count={count} profileIcons={profileIcons} onPressFollow={onPressFollow} is_following={is_following} />
      </View>
    )
  })
  return (
    <View style={styles.container}>
      {Platform.OS === 'ios' ? <View style={styles.topCover} /> : null}
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Image source={backIcon} style={styles.backIcon} />
        </TouchableOpacity>
        <Text style={styles.relatedText}>Related Entities</Text>
      </View>
      <ScrollView showsVerticalScrollIndicator={false}>
        <View style={styles.scrollContainer}>{EntitiesDataElements}</View>
      </ScrollView>
    </View>
  )
}

// relatedEntities.defaultProps = {
//   Data: [
//     {
//       title: 'India',
//       count: 52,
//       profileIcons: [images.EntitiesImage0]
//     },
//     {
//       title: 'United States of America',
//       count: 52,
//       profileIcons: [images.EntitiesImage1]
//     },
//     {
//       title: 'India',
//       count: 52,
//       profileIcons: [images.EntitiesImage0]
//     },
//     {
//       title: 'United States of America',
//       count: 52,
//       profileIcons: [images.EntitiesImage1]
//     },
//     {
//       title: 'India',
//       count: 52,
//       profileIcons: [images.EntitiesImage0]
//     },
//     {
//       title: 'United States of America',
//       count: 52,
//       profileIcons: [images.EntitiesImage1]
//     },
//     {
//       title: 'India',
//       count: 52,
//       profileIcons: [images.EntitiesImage0]
//     },
//     {
//       title: 'United States of America',
//       count: 52,
//       profileIcons: [images.EntitiesImage1]
//     }
//   ]
// }

export default withApollo(relatedEntities)
