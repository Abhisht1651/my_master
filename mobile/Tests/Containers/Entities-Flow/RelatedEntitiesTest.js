import React from 'react'
import { shallow } from 'enzyme'
import { Image } from 'react-native'
import { TouchableOpacity, Text } from 'react-native'
import RelatedEntities from '../../../App/Containers/EntitiesScreens/RelatedEntities'
// Test for Entities (snapshot and props)
describe('RelatedEntities', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<RelatedEntities route={{ params: { selected: [] } }} />) // no compile-time error
  })
  it('Related Entites component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
  })
  it('should render 1 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(1)
  })
})
