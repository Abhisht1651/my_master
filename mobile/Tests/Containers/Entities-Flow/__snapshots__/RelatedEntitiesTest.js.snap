// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedEntities Related Entites component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": 100.21428571428572,
        "justifyContent": "space-between",
        "paddingLeft": 53.99999999999999,
        "paddingRight": 229.99999999999997,
        "shadowColor": "#E6ECED",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
      />
    </ForwardRef>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 40,
        }
      }
    >
      Related Entities
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E6ECED",
      }
    }
  >
    <ScrollViewMock>
      <View
        style={
          Object {
            "paddingBottom": 8.214285714285715,
            "paddingTop": 59.142857142857146,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="India"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={51}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="United States of America"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="United States of America"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 8.214285714285715,
          }
        }
      >
        <CardType1
          count={0}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          title="Default"
        />
      </View>
    </ScrollViewMock>
  </View>
</View>
`;
